src/huak/utils/shell.rs

```diff
@@ -0,0 +1,49 @@
+use std::path::Path;
+
+use crate::errors::{HuakError, HuakResult};
+
+/// Gets the name of the current shell.
+///
+/// Returns an error if it fails to get correct env vars.
+pub fn get_shell_name() -> HuakResult<String> {
+    let shell_path = get_shell_path()?;
+    let shell_name = Path::new(&shell_path)
+        .file_name()
+        .and_then(|name| name.to_str())
+        .map(|name| name.to_owned())
+        .ok_or_else(|| {
+            HuakError::InternalError("Shell path is invalid.".to_owned())
+        });
+
+    shell_name
+}
+
+/// Gets the path of the current shell from env var
+///
+/// Returns an error if it fails to get correct env vars.
+pub fn get_shell_path() -> HuakResult<String> {
+    let shell_path: String = if cfg!(windows) {
+        std::env::var("COMSPEC")?
+    } else if cfg!(unix) {
+        std::env::var("SHELL")?
+    } else {
+        unimplemented!("We don't know how to get current shell for your OS.")
```

Comment:
Could we create an error for this?
```suggestion
        return Err(HuakError:UnsupportedOS)
```
Or something along those lines?

Reply:
Message would be "OS is unsupported for this action." or something.

Reply:
Checked it now and I think only wasm will end up in that branch. Not sure if we're planning to support wasm, or even if it's possible. 
But that's irrelevant to that. Turning this to an error makes sense

Reply:
Or maybe we should have separate functions, with `#[cfg(unix)]` and `#[cfg(windows)]` attributes, so it won't even compile on unsupported OSes. What do you think?

Reply:
Let's do it!